frontend/main-app/src/components/FormViewer/FormViewerUtils.tsx

```diff
@@ -0,0 +1,86 @@
+import { z, ZodNumber, ZodOptional, ZodString } from "zod";
+
+export const cleanFormData = (formData: FormData) => {
+  const obj = Object.fromEntries(formData);
+  const newObj = {};
+
+  for (const key in obj) {
+    const value = obj[key];
+    if (value === "") {
+      newObj[key] = undefined;
+    } else {
+      newObj[key] = value;
+    }
+  }
+
+  return newObj;
+};
+
+export const generateValidationSchema = (form) => {
+  const schemaShape = {};
+
+  form.components.forEach((component) => {
```

Comment:
What do you think about extracting the logic below this in two separated functions?
Then the `generateVaildationSchema` can look like something like this:
```
export const generateValidationSchema = (form): ZObject => {
  const schemaShape: Record<string, ZodString | ZodNumber> = {};

  form.components.forEach((component) => {
    let validator = z.string(); // Default for unknown types

    if (component.inputType === "string") {
      validator = getStringValidator(component);
    } else if (component.inputType === "number") {
      validator = getNumberValidator(component);
    }

    schemaShape[component.id] = validator;
  });

  return z.object(schemaShape);
    };
```